### PR TITLE
types: fixed unreachable overloads

### PR DESCRIPTION
**Pull request by kyranet**
**Please describe the changes this PR makes and why it should be merged:**

This PR resolves some unreachable method overloads in some classes, which caused some type inaccuracies.

cc: @ckohen 

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Review by iCrawl — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Comment by iCrawl** [comment]
This needs a rebase.

**Comment by kyranet** [comment]
Done, @iCrawl

File: typings/index.d.ts [hunk 1]
```diff
@@ -2071,19 +2071,15 @@ export class ApplicationCommandManager<
     null
   >;
   private commandPath({ id, guildId }: { id?: Snowflake; guildId?: Snowflake }): unknown;
+  public create(command: ApplicationCommandData): Promise<ApplicationCommandType>;
   public create(command: ApplicationCommandData, guildId: Snowflake): Promise<ApplicationCommand>;
-  public create(command: ApplicationCommandData, guildId?: Snowflake): Promise<ApplicationCommandType>;
   public delete(command: ApplicationCommandResolvable, guildId?: Snowflake): Promise<ApplicationCommandType | null>;
+  public edit(command: ApplicationCommandResolvable, data: ApplicationCommandData): Promise<ApplicationCommandType>;
```
Comment (DTrombett) [inline typings/index.d.ts:2077]: Only to me this line seems too long 😅? Did you try to run prettier?

Comment (kyranet) [reply typings/index.d.ts:2077]: It's the same length as the line above, which wasn't modified.

**Pull request merged**